MobileWallet/Screens/Send/AddNote/AddNoteViewController.swift

```diff
@@ -0,0 +1,268 @@
+	derived from this software without specific prior written permission.
+
+	THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+	CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+	OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+	DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+	CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+	SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+	NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+	LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+	HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+	OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+	SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import UIKit
+
+class AddNoteViewController: UIViewController, UITextViewDelegate, SlideViewDelegate {
+    var publicKey: PublicKey?
+    var amount: MicroTari?
+    private let SIDE_PADDING = Theme.shared.sizes.appSidePadding
+
+    fileprivate let sendButton = SlideView()
+    fileprivate var sendButtonBottomConstraint = NSLayoutConstraint()
+    fileprivate let titleLabel = UILabel()
+    fileprivate let noteInput = UITextView()
+    fileprivate let notePlaceholder = UILabel()
+    fileprivate var noteText = "" {
+        didSet {
+            if noteText.isEmpty {
+                sendButton.isEnabled = false
+            } else {
+                sendButton.isEnabled = true
+            }
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setup()
+
+        hideKeyboardWhenTappedAroundOrSwipedDown()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        styleNavigatorBar(isHidden: false)
+
+        guard let pubKey = publicKey else { return }
+
+        do {
+            try showNavbarEmojies(pubKey)
+        } catch {
+            UserFeedback.shared.error(
+                title: NSLocalizedString("Public key error", comment: "Add amount view"),
+                description: NSLocalizedString("Failed to get Emoji ID from user's contact", comment: "Add amount view"),
+                error: error
+            )
+        }
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(moveSendButtonUp), name: UIResponder.keyboardWillShowNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(moveSendButtonDown), name: UIResponder.keyboardWillHideNotification, object: nil)
+
+        noteInput.becomeFirstResponder()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        hideNavbarEmojis()
+        NotificationCenter.default.removeObserver(self)
```

Comment:
this should be added to deinit() not viewWillDissapear. All the removeObservers should be added to deinit() in case there are other observers in viewWillDissapear.

Reply:
What happens when another VC is pushed to the stack then? The current VC will not be deinitialised yet, but `viewWillDisappear ` will execute. Would we want the observers to still be active even when this VC is not on top of the stack?

Reply:
Ok so I see after iOS 9 all observers are removed automatically. We're targeting iOS 13 so I can actually remove this all together 👍